commands/cloudapi-v6/group.go

```diff
@@ -255,19 +263,29 @@ func RunGroupUpdate(c *core.CommandConfig) error {
 }
 
 func RunGroupDelete(c *core.CommandConfig) error {
-	c.Printer.Verbose("Group with id: %v is deleting...", viper.GetString(core.GetFlagName(c.NS, cloudapiv6.ArgGroupId)))
-	if err := utils.AskForConfirm(c.Stdin, c.Printer, "delete group"); err != nil {
-		return err
-	}
-	resp, err := c.CloudApiV6Services.Groups().Delete(viper.GetString(core.GetFlagName(c.NS, cloudapiv6.ArgGroupId)))
-	if resp != nil {
-		c.Printer.Verbose(cloudapiv6.RequestTimeMessage, resp.RequestTime)
-	}
-	if err != nil {
-		return err
-	}
-	if err = utils.WaitForRequest(c, waiter.RequestInterrogator, printer.GetId(resp)); err != nil {
-		return err
+	var resp *resources.Response
+	allFlag := viper.GetBool(core.GetFlagName(c.NS, cloudapiv6.ArgAll))
+	groupId := viper.GetString(core.GetFlagName(c.NS, cloudapiv6.ArgGroupId))
+	if allFlag {
+		err := DeleteAllGroups(c)
+		if err != nil {
+			return err
+		}
+	} else {
+		if err := utils.AskForConfirm(c.Stdin, c.Printer, "delete group"); err != nil {
+			return err
+		}
+		c.Printer.Verbose("Starting deleting Group with id: %v...", groupId)
+		resp, err := c.CloudApiV6Services.Groups().Delete(groupId)
```

Comment:
Use = instead of := to not define resp as a new variable. It will not be recognized at the end of the current function.